Fix #13609 (#15567)
nim-lang/Nim: doc/manual.rst

  p(3, 4) # now valid

> however discardable pragma not working on template because template do AST substitute in place, for example

Reviewer: Typos, grammar.

Reviewer 2: ```suggestion
however the discardable pragma does not work on templates as templates substitute the AST in place. For example:
```

.. code-block:: nim
  {.push discardable .}
  template example(): string = "https://nim-lang.org"
  {.pop.}

  example()

> This template will resolve into "https://nim-lang.org" which is string literal type, and since {.discardable.} don't apply to literals, the compiler will error.

Reviewer 2: ```suggestion
This template will resolve into "https://nim-lang.org" which is a string literal and since {.discardable.} doesn't apply to literals, the compiler will error.
```

Author: Thank you for help fix typo and grammar. Apologize for the mistake, english is not my native language.
